src/main/java/com/mixpanel/android/viewcrawler/ViewCrawler.java

```diff
@@ -221,27 +226,14 @@ private void uninstallConnectionSensor(final Activity activity) {
         }
 
         private boolean isInEmulator() {
-            if (!Build.HARDWARE.equals("goldfish")) {
-                return false;
-            }
-
-            if (!Build.BRAND.startsWith("generic")) {
-                return false;
-            }
-
-            if (!Build.DEVICE.startsWith("generic")) {
-                return false;
-            }
-
-            if (!Build.PRODUCT.contains("sdk")) {
-                return false;
-            }
-
-            if (!Build.MODEL.toLowerCase(Locale.US).contains("sdk")) {
-                return false;
-            }
-
-            return true;
+            return Build.FINGERPRINT.startsWith("generic")
+                    || Build.FINGERPRINT.startsWith("unknown")
```

Comment:
On the stackoverflow post there are 4 people suggesting `Build.MANUFACTURER.equals("unknown")`. Also, this library does it as well: https://github.com/gingo/android-emulator-detector/blob/master/EmulatorDetectorProject/EmulatorDetector/src/main/java/net/skoumal/emulatordetector/EmulatorDetector.java#L41

I think its worth to include it.